ooniapi/services/oonirun/src/oonirun/routers/oonirun.py

```diff
@@ -0,0 +1,628 @@
+    is_background_run_enabled_default: bool = Field(
+        default=False,
+        title="if this test should be enabled by default for background runs",
+    )
+    is_manual_run_enabled_default: bool = Field(
+        default=False, title="if this test should be enabled by default for manual runs"
+    )
+
+
+class OONIRunLinkEngineDescriptor(BaseModel):
+    revision: str = Field(title="revision of the nettest descriptor")
+    nettests: List[OONIRunLinkNettest] = Field(default=[], title="list of nettests")
+    date_created: datetime = Field(title="date when the nettest list was created")
+
+
+class OONIRunLinkBase(BaseModel):
+    name: str = Field(
+        default="", title="name of the ooni run link", min_length=2, max_length=50
+    )
+    short_description: str = Field(
+        default="",
+        title="short description of the ooni run link",
+        min_length=2,
+        max_length=200,
+    )
+
+    description: str = Field(
+        default="", title="full description of the ooni run link", min_length=2
+    )
+    author: str = Field(
+        default="",
+        title="public email address of the author name of the ooni run link",
+        min_length=2,
+        max_length=100,
+    )
+
+    nettests: List[OONIRunLinkNettest]
+
+    name_intl: Optional[Dict[str, str]] = Field(
+        default=None,
+        description="name of the ooni run link in different languages",
+    )
+    short_description_intl: Optional[Dict[str, str]] = Field(
+        default=None,
+        description="short description of the ooni run link in different languages",
+    )
+    description_intl: Optional[Dict[str, str]] = Field(
+        default=None,
+        description="full description of the ooni run link in different languages",
+    )
+
+    @validator("name_intl", "short_description_intl", "description_intl")
+    def validate_intl(cls, v):
+        # None is also a valid type
+        if v is None:
+            return v
+        for value in v.values():
+            if len(value) < 2:
+                raise ValueError("must be at least 2 characters")
+        return v
+
+    icon: Optional[str] = Field(
+        default=None,
+        description="icon to use for the ooni run link",
+    )
+    color: Optional[str] = Field(
+        default=None,
+        description="color to use for the ooni run link as a hex value prefixed with #",
+        pattern="^#(?:[0-9a-fA-F]{6})$",
+    )
+    expiration_date: datetime = Field(
+        default_factory=lambda: utcnow_seconds() + timedelta(days=30 * 6),
+        description="future time after which the ooni run link will be considered expired and no longer editable or usable (defaults to 6 months from now)",
+    )
+
+
+class OONIRunLink(OONIRunLinkBase):
+    oonirun_link_id: str
+    date_created: datetime = Field(
+        description="time when the ooni run link was created"
+    )
+    date_updated: datetime = Field(
+        description="time when the ooni run link was created"
```

Comment:
something like that?
```suggestion
        description="time when the ooni run nettest was last updated"
```

Reply:
Good catch!